fenixedu-ist-teacher-evaluation/src/main/java/pt/ist/fenixedu/teacher/evaluation/ui/spring/CreditsManagementService.java

```diff
@@ -26,4 +42,49 @@ public void editCreditsPool(CreditsPoolBean creditsPoolBean) {
         }
 
     }
+
+    @Atomic(mode = TxMode.WRITE)
+    public Set<String> setExecutionCourseEffortRate(MultipartFile file) throws Exception {
+        Set<String> output = new HashSet<String>();
+        byte[] bytes = ByteStreams.toByteArray(file.getInputStream());
```

Comment:
You should avoid doing this, as transaction restarts will fail to read the file a second time. You should read the file outside of the write block, and just pass its contents in.